### PR TITLE
Adding conditional language server startup and passing in project from user workspace

### PR DESCRIPTION
**Pull request by gracegoo-stripe**
I also moved the language server client out of the stripeLanguageServer directory because I think it belongs in the extension code.

# Testing

https://user-images.githubusercontent.com/75757829/122124584-552a3880-cde4-11eb-8146-5b2cd3161b26.mp4

### CODE REVIEW

**Comment by gracegoo-stripe** [comment]
Same deal with the builds -- will not pass on other platforms until we update our executable acquisition logic.

File: src/languageServerClient.ts [hunk 1]
```diff
@@ -109,4 +120,41 @@ export class StripeLanguageClient {
...
     outputChannel.appendLine('C# language service is running.');
   }
+
+  /**
+   * Returns a solutions file or project file if it exists in the workspace.
+   *
+   * If the user is working with multiple workspaces that contain C# projects, we don't know which one to run the server on, so we'll return the first one we find.
+   * In the future, we may want to prompt the user to pick one, or expose a command that let's the user change the tracking project and restart.
+   *
+   * Returns null if none of the workspaces are .NET projects.
```
Comment (vcheung-stripe) [inline src/languageServerClient.ts:130]:
```suggestion
   * Returns [] if none of the workspaces are .NET projects.
```

File: test/suite/languageServerClient.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  teardown(() => {
+    sandbox.restore();
+  });
+
+  suite('getDotnetProjectFiles', () => {
+    const workspacePath = '/my/path';
+
+    test('returns empty when no workspaces', async () => {
+      sandbox.stub(vscode.workspace, 'workspaceFolders').value([]);
+      console.log('BAOASPJD');
```
Comment (vcheung-stripe) [inline test/suite/languageServerClient.test.ts:22]: Nit: remove console.log

File: test/suite/languageServerClient.test.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      sandbox.stub(vscode.workspace, 'workspaceFolders').value([
+        {
+          index: 0,
+          name: 'workspace',
+          uri: vscode.Uri.file(workspacePath),
+        },
+      ]);
+      const fileFilesStub = sandbox.stub(vscode.workspace, 'findFiles');
+      fileFilesStub.onCall(0).returns(Promise.resolve([]));
+      fileFilesStub.onCall(1).returns(Promise.resolve([vscode.Uri.file(csprojFile)]));
```
Comment (vcheung-stripe) [inline test/suite/languageServerClient.test.ts:61]: Might be worth commenting what the first and second calls correspond to. Or better, we could make `findFilesStub` return different values depending on the argument.

**Review by vcheung-stripe — APPROVED** [review]
Nice! Just a couple small things, but they're non-blocking.

File: test/suite/languageServerClient.test.ts (hunk 2 above)
Comment (gracegoo-stripe) [reply test/suite/languageServerClient.test.ts:22]: haha oops, thank you

File: src/languageServerClient.ts (hunk 1 above)
Comment (gracegoo-stripe) [reply src/languageServerClient.ts:130]: Good catch!

File: test/suite/languageServerClient.test.ts (hunk 3 above)
Comment (gracegoo-stripe) [reply test/suite/languageServerClient.test.ts:61]: Yeah, I tried the latter at first but I could not get that stub to be invoked when I provided the arguments. Let me give it another try and then I'll just leave some comments if that doesn't work out

**Comment by gracegoo-stripe** [comment]
Addressed comments, ptal @vcheung-stripe

**Review by vcheung-stripe — APPROVED** [review]

**Pull request merged**